packages/desktop-client/src/components/spreadsheet/index.ts

```diff
@@ -1,4 +1,35 @@
-// @ts-strict-ignore
 import { type Query } from 'loot-core/src/shared/query';
 
-export type Binding = string | { name: string; value?; query?: Query };
+export type SpreadsheetFieldTypes = {
+  account: {
+    // Common fields
+    'uncategorized-amount': number;
+    'uncategorized-balance': number;
+
+    // Account fields
+    balance: number;
+    'accounts-balance': number;
+    'budgeted-accounts-balance': number;
+    'offbudget-accounts-balance': number;
+    balanceCleared: number;
+    balanceUncleared: number;
+  };
+};
+
+export type SheetNames = keyof SpreadsheetFieldTypes & string;
+
+export type SheetFields<SheetName extends SheetNames> =
+  keyof SpreadsheetFieldTypes[SheetName] & string;
+
+export type Binding<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  SheetName extends SheetNames = any,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  FieldName extends SheetFields<SheetName> = any,
```

Comment:
See PR description; the approach is to use `any` for a slow rollout and revert the use of `any` after the last PR in the stack is merged (3 PRs, can all be merged around the same time).